hw4_1/hw4_1/Program.cs

```diff
@@ -0,0 +1,300 @@
+﻿namespace hw4_1
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            string path = "..\\..\\..\\..\\hw4_1.txt";
+            if (!File.Exists(path))
+            {
+                Console.WriteLine("Incorrect file path!");
+                return;
+            }
+
+            string expression;
+            using (StreamReader reader = new StreamReader(path))
+            {
+                if ((expression = await reader.ReadToEndAsync()) == null)
+                {
+                    Console.WriteLine("Empty file!");
+                    reader.Close();
+                    return;
+                }
+            }
+            
+            Console.WriteLine("Expression in file: " + expression);
+            try
+            {
+                ExpressionTree tree = new ExpressionTree(expression);
+            }

```

Comment:
```suggestion
            ExpressionTree tree;

            try
            {
                 tree = new ExpressionTree(expression);
            }
```
В этом случае дерево будет доступно после секции `try-catch`.